InnerEye/ML/dataset/full_image_dataset.py

```diff
@@ -259,18 +261,21 @@ def _load_dataset_sources(self) -> Dict[str, PatientDatasetSource]:
                                     local_dataset_root_folder=self.args.local_dataset,
                                     image_channels=self.args.image_channels,
                                     ground_truth_channels=self.args.ground_truth_ids,
-                                    mask_channel=self.args.mask_id
+                                    mask_channel=self.args.mask_id,
+                                    for_inference=self.for_inference
```

Comment:
"for_inference" is not an ideal choice of name: it describes its intended use, but not what it does. I'd suggest something like "allow_incomplete_labels" or its inverse "are_all_labels_required".